### ISSUE TITLE
Fine-tuned BB2 does not pick up context

### ISSUE THREAD

**Issue by fraferra** [facebookresearch/ParlAI#4080]
**Current objective:** re-training blenderbot2 400M params on a new context aware dataset (passing contextual information about the persona).

**Excepted output:** when passing context we are expecting the model to pick up on certain information. For example, if we are passing the following text, `your persona: my name is allie` or `partner's persona: my name is allie` , to the context, the model should pick the name and call itself as the name. This behaviour has been observed in the pre-trained Blenderbot2 400M, leading me to think that the way the custom chat_service agent and world are set up correctly since it picks up the context when training is done correctly.

**Actual output:** The actual behaviour of the fine tuned model is different from the pre-trained one, in fact when passing the same context text as mentioned above the bot believes that the speaker is actually named `allie`, hence it confuses the context as being the context of the human the bot is talking to.

**Training dataset**

The dataset has been modeled to reflect the same CONVAI2 dataset's structure.

Here a single episode from the re-formatted custom dataset (note: personas are for now fictional, only used so the model can pick up on the pattern):

1 your persona: my name is john
2 partner's persona: my name is annabelle
3 partner's persona: i like snowboarding, cooking, and traveling.
4 hi there! how you doing?  	hi, good morning! i am good thanks! 

Similarly to CONVAI2 the first sentence is the first speaker (the human in my scenario) and the second one is the bot . The input (written by the human) and the label are separated by a tab (\t). The numerical value signals the beginning and the end of an episode.

**Configs**

In terms of training, this is my current config:

```
parlai train_model --model projects.blenderbot2.agents.blenderbot2:BlenderBot2FidAgent \
--task customchat_persona \
--num_epochs 10 \
--memory-decoder-model-file "" \
--n-layers 12 \
--init-model zoo:blenderbot2/blenderbot2_400M/model --dict-file zoo:blenderbot2/blenderbot2_400M/model.dict --fp16 True \
--datatype train:stream --batchsize 1 --truncate 256 --embedding-size 1024 --ffn-size 4096 --dropout 0.0 \
--attention-dropout 0.0 --n-heads 16 --learn-positional-embeddings True --embeddings-scale True \
--variant prelayernorm --activation gelu --n-encoder-layers 12 \
--n-decoder-layers 12 --model-parallel True --generation-model bart \
--query-model bert_from_parlai_rag \
--label-truncate 128 \
--rag-model-type token --rag-retriever-type dpr \
--max-doc-token-length 64 --dpr-model-file zoo:hallucination/bart_rag_token/model --beam-size 10 \
--beam-min-length 20 --beam-context-block-ngram 3 --beam-block-ngram 3 --beam-block-full-context False \
--inference beam --fp16 True --fp16-impl mem_efficient --optimizer mem_eff_adam --learningrate 1e-05 \
--delimiter '\n' \
--rag-retriever-type dpr \
--history-add-global-end-token end --dict-tokenizer bytelevelbpe \
--warmup-updates 100 --search-query-generator-model-file zoo:blenderbot2/query_generator/model \
--search-query-generator-beam-min-length 2 \
--gold-document-titles-key select-docs-titles --insert-gold-docs True \
--model-file /home/ubuntu/ParlAI/data/models/customchat_persona_model_4/model \
--save-every-n-secs 600 \
--split-lines True \
--max-doc-token-length 256 \
--force-fp16-tokens True \
--dict-tokenizer gpt2 --lr-scheduler-patience 1 --warmup-updates -1
```

Could you someone point me out on why, given the same dataset structure from CONVAI2 and using the same task agent, the model is not able to pick up on context like the pre-trained BB2 400M?

Thanks!

**Comment by klshuster** [comment]
The `memory_key` is the key in the `Message` from which BB2 extracts memories (i.e., personas). If you supply in the text field during training, the default value is fine (which is `--memory-key full_text`; this just extracts from the conversational context). If this is how you are presenting persona lines to your model, you'll need to set `--memory-key full_text` during inference as well, so that BB2 can extract the memories accordingly.

**Comment by klshuster** [comment]
No problem; closing for now, please reopen if there are any further issues
